core/types/rollup_cost.go

```diff
@@ -114,6 +123,42 @@ func NewL1CostFunc(config *params.ChainConfig, statedb StateGetter) L1CostFunc {
 	}
 	forBlock := ^uint64(0)
 	var cachedFunc l1CostFunc
+	selectFunc := func(blockTime uint64) l1CostFunc {
+		// Note: the various state variables below are not initialized from the DB until this
+		// point to allow deposit transactions from the block to be processed first by state
+		// transition.  This behavior is consensus critical!
+		l1FeeScalars := statedb.GetState(L1BlockAddr, L1FeeScalarsSlot).Bytes()
+		l1BlobBaseFee := statedb.GetState(L1BlockAddr, L1BlobBaseFeeSlot).Big()
+		l1BaseFee := statedb.GetState(L1BlockAddr, L1BaseFeeSlot).Big()
```

Comment:
Nit: with this change, it reads the ecotone scalars and blob-basefee data, even in bedrock blocks. It doesn't affect consensus, but does add a little overhead.